superset/migrations/versions/2024-05-10_18-02_f84fde59123a_update_charts_with_old_time_comparison.py

```diff
@@ -85,7 +85,7 @@ def upgrade_comparison_params(slice_params: dict[str, Any]) -> dict[str, Any]:
     # Adjust adhoc_custom
     if "adhoc_custom" in params and params["adhoc_custom"]:
         adhoc = params["adhoc_custom"][0]  # As there's always only one element
-        if adhoc["comparator"] != "No filter":
```

Comment:
I'm confused, why would this fail if the value is `None`?

```python
>>> None != "No filter"
True
```

Reply:
it would enter this conditional and `start_date_offset.strftime("%Y-%m-%d")` would fail since get_since_until(adhoc["comparator"]) would return None as the start_date

Reply:
Ah, I see, the problem is down the line. Makes sense, thanks!